File: temsys/src/main/java/com/deltegui/temsys/sensors/domain/Sensor.java
package com.deltegui.temsys.sensors.domain;

import com.deltegui.temsys.reports.Report;

import java.util.List;
import java.util.stream.Collectors;

public final class Sensor {
    private String name;
    private String connectionType;
    private String connectionValue;
    private int upateInterval;
    private boolean deleted;

    public Sensor(String name, String connectionType, String connectionValue, int updateInterval, boolean deleted) {
        this.name = name;
        this.connectionValue = connectionValue;
        this.connectionType = connectionType;
        this.upateInterval = updateInterval;
        this.deleted = deleted;
    }

    public List<Report> filterReports(List<Report> reports) {
        return reports.stream()
                .filter(report -> report.comesFromSensor(this))
                .collect(Collectors.toList());
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getConnectionValue() {
        return connectionValue;
    }

    public void setConnectionValue(String connectionValue) {
        this.connectionValue = connectionValue;
    }

    public String getConnectionType() {
        return connectionType;
    }

    public void setConnectionType(String connectionType) {
        this.connectionType = connectionType;
    }

    public int getUpateInterval() {
        return upateInterval;
    }

    public void setUpateInterval(int upateInterval) {
        this.upateInterval = upateInterval;
    }

    public boolean isDeleted() {
        return deleted;
    }

    public void setDeleted(boolean deleted) {
        this.deleted = deleted;
    }
}

File: back-evo/domain/Sensor.cs
using System.Collections.Generic;
using Domain.Core;

namespace Domain {
    public interface ISensorConnector {
        public Maybe<List<Report>> ReadDataFrom(string connectionValue);
    }

    public class Sensor {
        public string Name { get; }
        public int UpdateInterval { get; set; }
        public bool Deleted { get; set; }
        public List<string> SupportedReports { get; set; }
        public string ConnectionValue { get; set; }

        public Sensor (
            string name,
            int updateInterval,
            bool deleted,
            List<string> supportedReports,
            string connValue
        ) {
            this.Name = name;
            this.UpdateInterval = updateInterval;
            this.Deleted = deleted;
            this.SupportedReports = supportedReports;
            this.ConnectionValue = connValue;
        }

        public Maybe<List<Report>> GetCurrentState(ISensorConnector connector) {
            for (int i = 0; i < 2; i++) {
                var reports = connector.ReadDataFrom(this.ConnectionValue);
                if (reports.IsFilled()) {
                    return reports;
                }
            }
            return Maybe<List<Report>>.Empty();
        }
    }
}
